src/data-operations/data-container.ts

```diff
@@ -0,0 +1,124 @@
+import { FilteringExpression, FilteringLogic } from "./filtering-expression.interface";
+import { FilteringCondition } from "./filtering-condition";
+import { FilteringState } from "./filtering-state.interface";
+import { IFilteringStrategy, FilteringStrategy } from "./filtering-strategy";
+
+import { SortingExpression, SortingDirection } from "./sorting-expression.interface";
+import {SortingState} from "./sorting-state.interface";
```

Comment:
I say, lets run TSLint, and to enable it for the repository now. If we do this later, it's going to be even more of a pain!